spot_driver_cpp/src/kinematic/kinematic_service.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2023 Boston Dynamics AI Institute LLC. All rights reserved.
+
+#include <spot_driver_cpp/kinematic/kinematic_service.hpp>
+
+#include <spot_driver_cpp/conversions/kinematic_conversions.hpp>
+#include <spot_driver_cpp/interfaces/rclcpp_logger_interface.hpp>
+#include <spot_driver_cpp/kinematic/kinematic_middleware_handle.hpp>
+
+namespace {
+constexpr auto kServiceName = "get_inverse_kinematic_solutions";
+}
+
+namespace spot_ros2::kinematic {
+KinematicService::KinematicService(std::shared_ptr<KinematicApi> kinematic_api,
+                                   std::shared_ptr<LoggerInterfaceBase> logger,
+                                   std::unique_ptr<MiddlewareHandle> middleware_handle)
+    : kinematic_api_{kinematic_api}, logger_{std::move(logger)}, middleware_handle_{std::move(middleware_handle)} {}
+
+void KinematicService::initialize() {
+  service_ = middleware_handle_->create_service(
+      kServiceName, [this](const std::shared_ptr<GetInverseKinematicSolutions::Request> request,
+                           std::shared_ptr<GetInverseKinematicSolutions::Response> response) {
+        this->get_solutions(request, response);
+      });
+}
+
+void KinematicService::get_solutions(const std::shared_ptr<GetInverseKinematicSolutions::Request> request,
```

Comment:
Is there a reason to make the request shared_ptr `const`?

Reply:
It is not mandatory, but it is a suggested practice when creating services in ROS2: https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Cpp-Service-And-Client.html
The const is just to prevent reassignment of the shared pointer.